src/iexamine.cpp

```diff
@@ -5048,110 +5047,111 @@ void iexamine::ledge( Character &you, const tripoint &examp )
 
     uilist cmenu;
     cmenu.text = _( "There is a ledge here.  What do you want to do?" );
+
+    // NOTE this menu is merged with the climb down menu, manage keys carefully.
     cmenu.addentry( ledge_peek_down, true, 'p', _( "Peek down." ) );
-    if( you.has_flag( json_flag_WALL_CLING ) ) {
-        cmenu.addentry( ledge_cling_down, true, 'C', _( "Crawl down." ) );
-    }
-    cmenu.addentry( ledge_climb_down, true, 'c', _( "Climb down." ) );
+    g->climb_down_menu_gen( examp, cmenu );
     cmenu.addentry( ledge_jump_across, jump_target_valid, 'j',
                     ( jump_target_valid ? _( "Jump across." ) : _( "Can't jump across (need a small gap)." ) ) );
     cmenu.addentry( ledge_fall_down, true, 'f', _( "Fall down." ) );
 
     cmenu.query();
 
     creature_tracker &creatures = get_creature_tracker();
-    switch( cmenu.ret ) {
-        case ledge_jump_across: {
-            // If player is grabbed, trapped, or somehow otherwise movement-impeded, first try to break free
-            if( !you.move_effects( false ) ) {
-                you.moves -= 100;
-                return;
-            }
+    if( g->climb_down_menu_pick( examp, cmenu.ret ) ) {
+        // This branch means the player chose some option generated by the climb menu.
+    } else switch( cmenu.ret ) {
```

Comment:
This means we don't need to them indent the whole switch another level. I can't tell if you made functional changes to the switch because it's also been indented in an entire level.
```suggestion
    if( g->climb_down_menu_pick( examp, cmenu.ret ) ) {
        // This branch means the player chose some option generated by the climb menu.
        return;
    }
    switch( cmenu.ret ) {
```